ui/src/app/pages/admin/admin-user.tsx

```diff
@@ -136,32 +145,134 @@ const AdminUser = withUrlParams()(class extends React.Component<Props, State> {
+        verifiedInstitutionsByShortname: institutions.reduce(
+          (accumulator, institution) => accumulator.set(institution.shortName, institution),
+          new Map<String, PublicInstitutionDetails>()
+        )
+      });
     } catch (error) {
       this.setState({institutionsLoadingError: 'Could not get list of verified institutions - please try again later'});
     }
   }
 
+  getRoleOptionsForProfile() {
+    const {updatedProfile: {verifiedInstitutionalAffiliation}, verifiedInstitutionOptions} = this.state;
+    const institutionShortName = verifiedInstitutionalAffiliation ? verifiedInstitutionalAffiliation.institutionShortName : '';
+    return getRoleOptions(verifiedInstitutionOptions, institutionShortName);
+  }
+
+  getInstitutionDropdownOptions() {
+    const {verifiedInstitutionOptions} = this.state;
+    return fp.map(
+      institution => {
+        return {label: institution.displayName, value: institution.shortName};
+      },
+      verifiedInstitutionOptions
+    );
+  }
+
+  isSaveDisabled(errors) {
+    const {oldProfile, updatedProfile} = this.state;
+    return fp.isEqual(oldProfile, updatedProfile) || errors;
+  }
+
+  setVerifiedInstitutionOnProfile(institutionShortName: string) {
+    const {verifiedInstitutionsByShortname} = this.state;
+    this.setState(
+      fp.set(
+          ['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionShortName'],
+        institutionShortName
+      )
+    );
+    this.setState(
+      fp.set(
+          ['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionDisplayName'],
+        verifiedInstitutionsByShortname.get(institutionShortName).displayName
+        )
+    );
+    this.setState(fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionRoleEnum'], undefined));
+    this.setState(fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionalRoleOtherText'], undefined));
+  }
+
+  setInstitutionalRoleOnProfile(institutionalRoleEnum: InstitutionalRole) {
+    this.setState(fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionalRoleEnum'], institutionalRoleEnum));
+    this.setState(fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionalRoleOtherText'], undefined));
+  }
+
+  updateVerifiedInstitutionalAffiliation() {
+    const {updatedProfile} = this.state;
+    this.setState({loading: true});
+    profileApi().updateVerifiedInstitutionalAffiliation(updatedProfile.userId, updatedProfile.verifiedInstitutionalAffiliation).then(() => {
+      this.setState({loading: false});
+    });
+  }
+
+  validateVerifiedInstitutionalAffiliation() {
+    const {updatedProfile} = this.state;
+    if (updatedProfile) {
+      if (updatedProfile.verifiedInstitutionalAffiliation) {
+        return updatedProfile.verifiedInstitutionalAffiliation;
+      }
+      return false;
+    }
+    return false;
```

Comment:
Take a look at [optional chaining ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) with a ternary here - it may clean this up a bit

If that works, it may work for a few of the functions below as well

Reply:
Our version of TypeScript doesn't support that.